src/transformers/generation/logits_process.py

```diff
@@ -172,11 +172,73 @@ def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> to
 
 class TemperatureLogitsWarper(LogitsWarper):
     r"""
-    [`LogitsWarper`] for temperature (exponential scaling output probability distribution).
+    [`LogitsWarper`] for temperature (exponential scaling output probability distribution) which effectively means that
+    we can control the randomness of the predicted tokens.
+
+    <Tip>
+
+    Make sure that `do_sample=True` is included in the `generate` arguments otherwise the temperature value won't have
+    any effect. As noted in `generation_strategies#Assisted-Decoding`, this parameter along with assisted decoding will
+    help improving the latency.
+
+    </Tip>
 
     Args:
         temperature (`float`):
-            The value used to module the logits distribution.
+            The value used to module the logits distribution. A value of `1.` will make no difference in the output
+            with respect to calling `generate` without temperature.
```

Comment:
```suggestion
            The value used to module the logits distribution. A value smaller than `1` decreases randomness (and vice versa), with `0` being equivalent to shifting all probability mass to the most likely token.
```

Let's focus on the relevant use cases (temperature < 1) rather than on the caveats :)